hibike/unit_tests/hibike_message_tests.py

```diff
@@ -0,0 +1,373 @@
+"""
+Unit tests for functions in hibike_message.
+"""
+
+
+import unittest
+import random
+import queue
+import time
+import threading
+
+import serial
+
+import hibike_message
+import spawn_virtual_devices
+from unit_tests.utils import run_with_random_data
+DEVICE_TYPES = list(hibike_message.DEVICES)
+
+
+def random_uid():
+    """
+    Generate a random UID.
+    """
+    uid_type = random.choice(DEVICE_TYPES)
+    return uid_type << 72 | random.getrandbits(72)
+
+
+def random_params(device_id):
+    """
+    Get a random sampling of parameters for a particular DEVICE_TYPE.
+    """
+    all_params = hibike_message.all_params_for_device_id(device_id)
+    sample_length = random.randrange(len(all_params) + 1)
+    params = random.sample(all_params, sample_length)
+    return params
+
+
+def random_values(device_id, params):
+    """
+    Generate random valid values for PARAMS.
+    """
+    all_params = hibike_message.DEVICES[device_id]["params"]
+    values = []
+    for param in params:
+        param_index = 0
+        for (idx, param_pack) in enumerate(all_params):
+            if param_pack["name"] == param:
+                param_index = idx
+                break
+        param_type = all_params[param_index]["type"]
```

Comment:
I'd add an if statement here: 
`if param_index == 0`
better to be explicit, and more readable
add a corresponding `else` case that should never happen, and just throw a generic exception